### ISSUE TITLE
PHP 8.1 incompatibility

### ISSUE THREAD

**Issue by RedSign77** [tcdent/php-restclient#41]
Hey!

Magento 2 uses the tcdent/php-restclient package and the new version (2.4.4) has a lot of PHP 8.1 incompatible issues:

```PHP Deprecated:  Return type of RestClient::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 66

Deprecated: Return type of RestClient::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 66
PHP Deprecated:  Return type of RestClient::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 74

Deprecated: Return type of RestClient::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 74
PHP Deprecated:  Return type of RestClient::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 70

Deprecated: Return type of RestClient::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 70
PHP Deprecated:  Return type of RestClient::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 78

Deprecated: Return type of RestClient::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 78
PHP Deprecated:  Return type of RestClient::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 61

Deprecated: Return type of RestClient::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 61
PHP Deprecated:  Return type of RestClient::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 84

Deprecated: Return type of RestClient::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 84
PHP Deprecated:  Return type of RestClient::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 90

Deprecated: Return type of RestClient::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 90
PHP Deprecated:  Return type of RestClient::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 99

Deprecated: Return type of RestClient::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 99
PHP Deprecated:  Return type of RestClient::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 103

Deprecated: Return type of RestClient::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/html/vendor/tcdent/php-restclient/restclient.php on line 103
```

**Comment by spider4216** [comment]
the same issue :(

**Comment by RedSign77** [comment]
Hey @tcdent !
I'm solved locally the incompatibility issues, here is the patch file:
[php8.1-incompatibility-fix.patch.tar.gz](https://github.com/tcdent/php-restclient/files/8424290/php8.1-incompatibility-fix.patch.tar.gz)

**Comment by spider4216** [comment]
I apply you patch in my project but can't run tests because PHPUnit tests and PHP 8 incompatible. Need rewrite tests on latest PHPUnit...

Also I send [pull request ](https://github.com/tcdent/php-restclient/pull/40) in the begging of year but no response yet...

Also I sent email to author but no response :(

![image](https://user-images.githubusercontent.com/10693428/162351339-467a9f2d-4924-4b55-9772-ae7ff9ceb182.png)

It seems to be the project is not supported anymore :'(

I was forced to move my projects to [Guzzle, PHP HTTP client](https://github.com/guzzle/guzzle)

**Comment by RedSign77** [comment]
Hey @spider4216 !

Yeah, it's seems to be an abandoned package. :(  I really hope everything is good with Travis!

For Magento 2 Adobe needs to replace the package :( 

Thank you,
Zoltán

**Comment by tcdent** [comment]
Hey guys,

Thrilled you still find a need for this library nearly 10 years in.

Does anyone have time to rewrite the tests as well? That would make it much easier for me to merge the compatibility fixes ensure a stable user experience.

**Comment by tcdent** [comment]
Just released 0.1.8 thanks to your input. All test are passing!
